### PR TITLE
workaround loginsstorage exception on first-time login

### LINKED ISSUE #244: Sync is not working on initial sign in
## Steps to reproduce

1. Open Lockbox on any device
2. Tap on "Get Started"
3. Login with a valid username and password.
4. Wait for sync

### Expected behaviour
Sync is performed and a list of logins should populate the "All entries" screen

### Actual behaviour
No sync is performed

### Device and build information
- Nexus 5 (Android 6.0.1)
- HTC Desire 820 (Android 6.0.1)
- Nokia 6 (Android 7.1.1)
- OnePlus 5T (Android 8.1.0)
- Google Pixel (Android 9)
- [Build #1316](https://app.bitrise.io/build/67245d12e783e79d)

### Notes

- Nothing was synced even after 20 minutes of waiting
- [Logcat](https://github.com/mozilla-lockbox/lockbox-android/files/2624537/log.txt)
- screen

### PR DESCRIPTION
**Pull request by sashei**
Closes #244 (again)

Connected to #240

## Testing and Review Notes

This should resolve the issue surfaced via autolock testing on #240 in which first-time login would not trigger a sync.

## To Do

- [x] add testing notes and screenshots in PR description to help guide reviewers
- [ ] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

**Comment by sashei** [comment]
@devinreams I was seeing the lock now issues on the autolock branch as well, might be worth checking against master to file another issue before blocking here. I'll look as well.

**Comment by sashei** [comment]
@devinreams ....not happening for me on master, or on this branch.... pixel 27 problem again???

**Comment by devinreams** [comment]
>The app bypassed the "Unlock" screen and dropped me right into syncing and a populated list view.

Ugh, not happening for me anymore, either. Sometimes my bugs have bugs. 🐛

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 1]
```diff
@@ -159,7 +159,8 @@ open class DataStore(
     }
 
     private fun reset() {
-        if (stateSubject.value == State.Unprepared) {
+        if (stateSubject.value == State.Unprepared ||
+            stateSubject.value == null) {
```
Comment (linuxwolf) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: 🤦‍♂️  nice catch

**Comment by devinreams** [comment]
@sashei I re-ran the tests but looks like UI tests are truly failing on the settings test (looking for the Get Started button?) am I seeing that right? 🙃

**Comment by sashei** [comment]
@devinreams yes that's first on my list for today.

**Pull request merged**